ui/ducks/bridge/selectors.ts

```diff
@@ -140,48 +167,196 @@ export const getBridgeQuotesConfig = (state: BridgeAppState) =>
     BridgeFeatureFlagsKey.EXTENSION_CONFIG
   ] ?? {};
 
+const _getBridgeFeesPerGas = createSelector(
+  getGasFeeEstimates,
+  (gasFeeEstimates) => ({
+    estimatedBaseFeeInDecGwei: (gasFeeEstimates as GasFeeEstimates)
+      ?.estimatedBaseFee,
+    maxPriorityFeePerGasInDecGwei: (gasFeeEstimates as GasFeeEstimates)?.[
+      BRIDGE_PREFERRED_GAS_ESTIMATE
+    ]?.suggestedMaxPriorityFeePerGas,
+    maxFeePerGas: decGWEIToHexWEI(
+      (gasFeeEstimates as GasFeeEstimates)?.high?.suggestedMaxFeePerGas,
+    ),
+    maxPriorityFeePerGas: decGWEIToHexWEI(
+      (gasFeeEstimates as GasFeeEstimates)?.high?.suggestedMaxPriorityFeePerGas,
+    ),
+  }),
+);
+
+export const getBridgeSortOrder = (state: BridgeAppState) =>
+  state.bridge.sortOrder;
+
+// A dest network can be selected before it's imported
+// The cached exchange rate won't be available so the rate from the bridge state is used
+const _getToTokenExchangeRate = createSelector(
+  (state) => state.metamask.currencyRates,
+  (state: BridgeAppState) => state.bridge.toTokenExchangeRate,
+  getToChain,
+  getToToken,
+  (cachedCurrencyRates, toTokenExchangeRate, toChain, toToken) => {
+    return (
+      toTokenExchangeRate ??
+      (isNativeAddress(toToken?.address) && toChain?.nativeCurrency
+        ? cachedCurrencyRates[toChain.nativeCurrency]?.conversionRate
+        : null)
+    );
+  },
+);
+
+const _getQuotesWithMetadata = createDeepEqualSelector(
+  (state) => state.metamask.bridgeState.quotes,
+  _getToTokenExchangeRate,
+  (state: BridgeAppState) => state.bridge.fromTokenExchangeRate,
+  getConversionRate,
+  _getBridgeFeesPerGas,
+  (
+    quotes,
+    toTokenExchangeRate,
+    fromTokenExchangeRate,
+    nativeExchangeRate,
+    { estimatedBaseFeeInDecGwei, maxPriorityFeePerGasInDecGwei },
+  ): (QuoteResponse & QuoteMetadata)[] => {
+    const newQuotes = quotes.map((quote: QuoteResponse) => {
+      const toTokenAmount = calcToAmount(quote.quote, toTokenExchangeRate);
+      const totalNetworkFee = calcTotalNetworkFee(
+        quote,
+        estimatedBaseFeeInDecGwei,
+        maxPriorityFeePerGasInDecGwei,
+        nativeExchangeRate,
+      );
+      const sentAmount = calcSentAmount(
+        quote.quote,
+        isNativeAddress(quote.quote.srcAsset.address)
+          ? nativeExchangeRate
+          : fromTokenExchangeRate,
+      );
+      const adjustedReturn = calcAdjustedReturn(
+        toTokenAmount.fiat,
```

Comment:
What unit is the fiat measured in? I think a comment should be left in the types. Like if it's USD, or the user selected fiat currency etc...